src/components/Game/Timer/Timer.js

```diff
@@ -2,58 +2,58 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
 import classes from './Timer.scss';
-import { computeMinutes, computeSeconds, printTwoDigits } from '../../../helpers';
+import { formatTimeLeft } from '../../../helpers';
+import { DURATION, INTERVAL } from '../../../constants/constants';
+
 
 class Timer extends Component {
     state = {
-        // 60 sec. only for testing reasons. it's gonna be more :)
-        seconds: 60,
+        timeLeft: DURATION,
     };
 
+    start = new Date().getTime();
+
     componentDidMount() {
+        this.setState({
+            timeLeft: this.getTimeLeft(new Date()),
+        });
         this.startTimer();
     }
 
     componentWillUnmount() {
         this.stopTimer();
     }
 
+    getTimeLeft(now) {
+        return DURATION - (now.getTime() - this.start);
+    }
+
     startTimer = () => {
         const { timeIsOver } = this.props;
 
         this.interval = setInterval(() => {
-            const { seconds } = this.state;
-
-            this.decrement();
-            if (seconds <= 1) {
-                this.stopTimer();
-                timeIsOver();
-            }
-        }, 1000);
-    };
-
-    decrement = () => {
-        this.setState((state) => ({
-            seconds: state.seconds - 1,
-        }));
+            this.setState({
+                timeLeft: this.getTimeLeft(new Date()),
+            }, () => {
```

Comment:
callback to `setState` function is called with new state as a first argument so you can have:
`({ timeLeft }) => {` and no line 38

Reply:
hmm, doesn't work 🤔 I get an error:
`TypeError: (destructured parameter) is undefined`

Reply:
oooops, I was wrong! callback is not passed any arguments, sorry, please ignore my comment then!